docs/standard/microservices-architecture/microservice-ddd-cqrs-patterns/implement-value-objects.md

```diff
@@ -120,25 +122,170 @@ public class Address : ValueObject
 }
 ```
 
-## Hiding the identity characteristic when using EF Core to persist value objects
+## How to persist value objects in the database with EF Core 2.0
+
+You just saw how to define a value object in your domain model. But, how can you actually persist it into the database through Entity Framework Core which usually targets entities with identity?
+
+### Background and older approaches using EF Core 1.1
+
+As background or history, a limitation when using EF Core 1.0 and 1.1 was that you cannot use  [complex types](https://docs.microsoft.com/de-de/dotnet/api/system.componentmodel.dataannotations.schema.complextypeattribute?view=netframework-4.7) as defined in EF 6.x in the traditional .NET Framework. Therefore, if using EF Core 1.0 or 1.1, you needed to store your value object as an EF entity with an ID field. Then, so it looked more like a value object with no identity, you could hide its ID so you make clear that the identity of a value object is not important in the domain model. You could hide that ID by using the ID as a [shadow property](https://docs.microsoft.com/ef/core/modeling/shadow-properties ). Since that configuration for hiding the ID in the model is set up in the EF infrastructure level, it would be kind of transparent for your domain model.
+
+In the initial version of eShopOnContainers (.NET Core 1.1), the hidden ID needed by EF Core infrastructure was implemented in the following way in the DbContext level, using Fluent API at the infrastructure project. Therefore, the ID was hidden from the domain model point of view, but still present in the infrastructure.
+
+```csharp
+// Old approach with EF Core 1.1
+// Fluent API within the OrderingContext:DbContext in the Infrastructure project
+void ConfigureAddress(EntityTypeBuilder<Address> addressConfiguration) 
+{
+    addressConfiguration.ToTable("address", DEFAULT_SCHEMA); 
+
+    addressConfiguration.Property<int>("Id")  // Id is a shadow property
+        .IsRequired();
+    addressConfiguration.HasKey("Id");   // Id is a shadow property
+}
+```
+
+However, the persistence of that value object into the database was performed like a regular entity in a different table.
+
+With EF Core 2.0 there are new and better ways to persist value objects.
+
+## Persist value objects as owned entity types in EF Core 2.0
+
+Even when there are still some gaps between the canonical value object pattern in DDD and the owned entity type in EF Core (see limitations at the end of this section), it is currently, with EF Core 2.0, the best way to persist value objects.
```

Comment:
the flow can be improved here. Maybe

Even with some gaps between the canonical value object pattern in DDD and the owned entity type in EF Core, it's currently the best way to persist value objects with EF Core 2.0.  You can see limitations at the end of this section.

Reply:
👍